alfalfa_web/server/api-v2.js

```diff
@@ -0,0 +1,408 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+ *  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ *  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ***********************************************************************************************************************/
+
+import AWS from "aws-sdk";
+import { Router } from "express";
+import got from "got";
+import { make, regex } from "simple-body-validator";
+import { version } from "../package.json";
+
+AWS.config.update({ region: process.env.REGION || "us-east-1" });
+const sqs = new AWS.SQS();
+
+let db;
+const router = Router();
+
+const api = ({ db: _db }) => {
+  db = _db;
+  return router;
+};
+
+router.get("/", (req, res) => {
+  res.redirect(301, "/docs");
+});
+
+function mapHaystack(row) {
+  return Object.keys(row).reduce((result, key) => {
+    result[key] = row[key].replace(/^[a-z]:/, "");
+    return result;
+  }, {});
+}
+
+/**
+ * @openapi
+ * /models:
+ *   get:
+ *     description: Return all models
+ *     operationId: models
+ *     tags:
+ *       - Alfalfa
+ *     responses:
+ *       200:
+ *         description: models response
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 $ref: '#/components/schemas/Model'
+ *             example:
+ *               data:
+ *                 - id: 4b8c6b40-f818-11ec-a8bd-75570e3e3a28
+ *                   description: Building 1
+ *                   uploadTimestamp: '2022-06-30 01:59:42.100453+00:00'
+ *                   uploadPath: uploads/4b8c6b40-f818-11ec-a8bd-75570e3e3a28/refrig_case_osw.zip
+ *                 - id: 82ae8d50-f837-11ec-bda1-355419177ef9
+ *                   description: Building 2
+ *                   uploadTimestamp: '2022-06-30 05:43:06.884923+00:00'
+ *                   uploadPath: uploads/82ae8d50-f837-11ec-bda1-355419177ef9/refrig_case_osw_2.zip
+ */
+router.get("/models", async (req, res) => {
+  const docs = {};
+  const cursor = db.collection("runs").find();
+  for await (const doc of cursor) {
+    if (doc) docs[doc._id] = doc;
+  }
+
+  got
+    .post("http://localhost/haystack/read", {
```

Comment:
My feeling is that the new api should not use the haystack api to query data. It should just go straight to mongo itself.